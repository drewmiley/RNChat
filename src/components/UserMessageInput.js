import React, { Component } from 'react';
import {
    Text,
    TextInput,
    TouchableHighlight,
    View
} from 'react-native';

export default class UserMessageInput extends React.Component {
    constructor(props) {
        super(props);
        this.shouldComponentUpdate = false;
        this.state = { text: 'Send Drew a message!' };
    }
    sendMessage() {
        this.props.userSendsMessage(this.state.text);
        this.setState({ text: 'Send Drew a message!' });
    }
    render() {
        return <View>
            <TextInput value={this.state.text} onChangeText={(text) => {this.setState({text})}} />
            <TouchableHighlight onPress={() => this.sendMessage()}>
                <Text>Send</Text>
            </TouchableHighlight>
        </View>
    }
};