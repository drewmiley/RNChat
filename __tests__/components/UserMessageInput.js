import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import UserMessageInput from '../..//src/components/UserMessageInput';

describe('UserMessageInput', () => {

    describe('Rendering', () => {

        it('should match the snapshot', () => {
            const component = renderer.create(
                <UserMessageInput />
            );            
            expect(component).toMatchSnapshot();
        });

    });

});