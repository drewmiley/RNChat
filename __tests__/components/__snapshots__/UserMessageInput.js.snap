exports[`UserMessageInput Rendering should match the snapshot 1`] = `
<View>
  <TextInput
    onChangeText={[Function]}
    value="Send Drew a message!" />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Send
    </Text>
  </View>
</View>
`;
